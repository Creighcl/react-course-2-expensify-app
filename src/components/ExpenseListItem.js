import React from 'react';
import { removeExpense } from '../actions/expenses';
import { NavLink } from 'react-router-dom';

export default ({ dispatch, id, description, amount, createdAt }) => (
  <div>
    <NavLink to={`/edit/${id}`}>{description}</NavLink>
    <p>{amount} - {createdAt}</p>
  </div>
);

