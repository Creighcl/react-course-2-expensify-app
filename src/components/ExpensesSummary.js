import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => (
    <div>
        Viewing { props.expensesCount } expense{ props.expensesCount === 1 ? '' : 's' } totalling { props.expenseTotal }    
    </div>
);

const mapStateToProps = (state) => {
    const currentExpenses = selectExpenses(state.expenses, state.filters);
    const expenseTotal = numeral(expensesTotal(currentExpenses)/100).format('$0,0.00');
    return {
        expensesCount: currentExpenses.length,
        expenseTotal
    };
};

export default connect(mapStateToProps)(ExpensesSummary);