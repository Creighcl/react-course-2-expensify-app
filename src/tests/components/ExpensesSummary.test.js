import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
import { filters, altFilters } from '../fixtures/filters';
import { shallow } from 'enzyme';

test('should match snapshot for std filters, expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={expenses.length } expenseTotal={'$100.00'} />);
    expect(wrapper).toMatchSnapshot();
});

test('should match snapshot for std filters, 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={'$100.00'} />);
    expect(wrapper).toMatchSnapshot();
});