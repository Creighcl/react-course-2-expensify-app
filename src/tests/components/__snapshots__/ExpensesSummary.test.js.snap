// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot for std filters, 1 expense 1`] = `
<div>
  Viewing 
   expense
  s
   totalling 
  $100.00
</div>
`;

exports[`should match snapshot for std filters, expenses 1`] = `
<div>
  Viewing 
   expense
  s
   totalling 
  $100.00
</div>
`;
